insights.go

```diff
@@ -17,6 +19,10 @@ func unregisterInsights() error {
 	return cmd.Run()
 }
 
+// insightsIsRegistered tries to get information if system is registered
+// using insights-client, and it can also return error. Current implementation
+// of insights-client is suboptimal. For this reason getting status of
+// insights-client is horrible heuristics. Hopefully, it will change in the future.
```

Comment:
Could we drop the emotions here a bit? :)

> `...using insights-client. The package is not able to reliably communicate its status via return codes. We have to use its semi-private hidden files to determine the actual status.`

Reply:
No, we cannot! Emotions are spice of life.

Reply:
...and they might be very confusing to someone new just trying to figure out why the code is so weird here :)

Reply:
Opinion comments tend to draw attention away from the intention of the comment. Code that is non-obvious should be commented with a brief explanation for the deviation from the normal or expected pattern, and nothing more.

Comments tend to "bit rot" more than code itself because they have no influence over the behavior of the program. The more text present in a comment, the more chance there is for the comment to remain unmodified as the code around it changes, leading to comments that are ignored by developers, but left in the code base to mislead new developers in the future.

I try to comment code as little as possible, and instead try to write the code so that it documents itself.